### PR TITLE
[LETS-215] Separate transaction server handling on active/passive branches

### PR DESCRIPTION
**Pull request by adrianbrab**
http://jira.cubrid.org/browse/LETS-215

Transformed the global variable holding the ATS  into a TS pointer that can be initialized as either a ATS or PTS.

### CODE REVIEW

File: src/storage/slotted_page.c [hunk 1]
```diff
@@ -5276,7 +5276,5 @@ spage_need_compact (THREAD_ENTRY * thread_p, PAGE_PTR page_p)
 bool
 spage_is_save_space_disabled ()
 {
-  const bool during_crash_recovery = log_is_in_crash_recovery ();
-  const bool in_page_server (get_server_type () == SERVER_TYPE_PAGE);
-  return during_crash_recovery || in_page_server;
+  return log_is_in_crash_recovery () || is_passive_server ();
```
Comment (cristiarg) [inline src/storage/slotted_page.c]: The extra variables were added to ease [conditional] debugging.
They might look ugly but the generated code is the same - ie: they'll be optimized away by the compiler in optimized build configs.

**Review by cristiarg — APPROVED** [review]

**Comment by ribram** [comment]
Fix build issues for SA_MODE:
```
../sa/libcubridsa.so.12.0: undefined reference to `is_active_transaction_server()'
../sa/libcubridsa.so.12.0: undefined reference to `is_passive_server()'
```
Define the functions in SA_MODE; stand-alone mode is considered an active transaction server.

File: src/base/server_type.cpp [hunk 2]
```diff
@@ -125,7 +126,21 @@ int init_server_type (const char *db_name)
...
+	}
+      else if (is_passive_transaction_server())
+	{
+	  ts_Gl.reset (new passive_tran_server());
+	}
+      else
+	{
+	  assert (false);
+	}
+      er_code = ts_Gl->boot (db_name);
```
Comment (ribram) [inline src/base/server_type.cpp]: Space after function name.
```suggestion
      if (is_active_transaction_server ())
	{
	  ts_Gl.reset (new active_tran_server ());
	}
      else if (is_passive_transaction_server ())
	{
	  ts_Gl.reset (new passive_tran_server ());
	}
      else
	{
	  assert (false);
	}
      er_code = ts_Gl->boot (db_name);
```

File: src/storage/slotted_page.c (hunk 1 above)
Comment (adrianbrab) [reply src/storage/slotted_page.c]: i changed them back to variables

**Review by ribram — APPROVED** [review]

File: src/transaction/boot_sr.c [hunk 3]
```diff
@@ -2659,7 +2659,7 @@ boot_restart_server (THREAD_ENTRY * thread_p, bool print_restart, const char *db
 #endif
 
 #if defined (SERVER_MODE)
-  if (get_server_type () == SERVER_TYPE_TRANSACTION)
+  if (is_active_transaction_server ())
```
Comment (hornetmj) [inline src/transaction/boot_sr.c:2662]: Compared to the develop branch, PTS also seems to have to call css_set_ha_num_of_hosts() and css_change_ha_server_state().

**Pull request merged**